benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -243,3 +265,109 @@ fn none_changed_detection(criterion: &mut Criterion) {
         );
     }
 }
+fn insert_if_bit_enabled<const B: u16>(entity: &mut EntityWorldMut, i: u16) {
+    if i & 1 << B != 0 {
+        entity.insert(Data::<B>(1.0));
+    }
+}
+fn add_archetypes_entites<T: Component + Default>(
+    world: &mut World,
+    archetype_count: u16,
+    entity_count: u32,
+) {
+    for i in 0..archetype_count {
+        for _j in 0..entity_count {
+            let mut e = world.spawn(T::default());
+            insert_if_bit_enabled::<0>(&mut e, i);
+            insert_if_bit_enabled::<1>(&mut e, i);
+            insert_if_bit_enabled::<2>(&mut e, i);
+            insert_if_bit_enabled::<3>(&mut e, i);
+            insert_if_bit_enabled::<4>(&mut e, i);
+            insert_if_bit_enabled::<5>(&mut e, i);
+            insert_if_bit_enabled::<6>(&mut e, i);
+            insert_if_bit_enabled::<7>(&mut e, i);
+            insert_if_bit_enabled::<8>(&mut e, i);
+            insert_if_bit_enabled::<9>(&mut e, i);
+            insert_if_bit_enabled::<10>(&mut e, i);
+            insert_if_bit_enabled::<11>(&mut e, i);
+            insert_if_bit_enabled::<12>(&mut e, i);
+            insert_if_bit_enabled::<13>(&mut e, i);
+            insert_if_bit_enabled::<14>(&mut e, i);
+            insert_if_bit_enabled::<15>(&mut e, i);
+        }
+    }
+}
+fn multiple_archetype_none_changed_detection_generic<T: Component + Default + BenchModify>(
+    group: &mut BenchGroup,
+    archetype_count: u16,
+    entity_count: u32,
+) {
+    group.bench_function(
+        format!(
+            "{}_archetypes_{}_entities_{}",
+            archetype_count,
+            entity_count,
+            std::any::type_name::<T>()
+        ),
+        |bencher| {
+            bencher.iter_batched_ref(
+                || {
+                    let mut world = World::new();
+                    add_archetypes_entites::<T>(&mut world, archetype_count, entity_count);
+                    world.clear_trackers();
+                    let mut query = world.query::<(
+                        Option<&mut Data<0>>,
+                        Option<&mut Data<1>>,
+                        Option<&mut Data<2>>,
+                        Option<&mut Data<3>>,
+                        Option<&mut Data<4>>,
+                        Option<&mut Data<5>>,
+                        Option<&mut Data<6>>,
+                        Option<&mut Data<7>>,
+                        Option<&mut Data<8>>,
+                        Option<&mut Data<9>>,
+                        Option<&mut Data<10>>,
+                        Option<&mut Data<11>>,
+                        Option<&mut Data<12>>,
+                        Option<&mut Data<13>>,
+                        Option<&mut Data<14>>,
+                    )>();
+                    for components in query.iter_mut(&mut world) {
+                        modify!(components;0,1,2,3,4,5,6,7,8,9,10,11,12,13,14);
```

Comment:
Sorry for the noob question; but why doesn't this trigger change detection?
I thought change detection was triggered on any DerefMut, even if the value itself is unchanged

Reply:
Oh I see you're querying for `Sparse` and `Table`, not `Data<X>`
Maybe we could have a comment explaining what the benchmark is doing